### DISCUSSION TITLE
help: use of `emit`

### DISCUSSION THREAD

**Discussion by milkliker** [glideapps/glide-data-grid#489] [Q&A] [answered]
Hi,

I want to add custom context menu of "copy" and "paste",
and when the menu btn clicked I called the `ref.current.emit('copy')` and `ref.current.emit('paste')`, nothing happens.

Do I misunderstand something? Thanks

**Comment by jassmith** [comment, marked as answer]
As both of those events interact with the clipboard you will need to ensure that they are invoked within an input event callback to make sure they work. The browser may otherwise choose to ignore the data grid getting/setting the clipboard.

**Reply by milkliker** [reply]
I see, thanks a lot!
